servers/src/main/java/tachyon/worker/block/BlockMetadataView.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.StorageTierView;
+
+/**
+ * This class exposes a narrower view of {@link BlockMetadataManager} to Evictors and Allocators.
+ */
+public class BlockMetadataView {
+
+  private final BlockMetadataManager mMetadataManager;
+  private List<StorageTierView> mTierViews = new ArrayList<StorageTierView>();
+  private final List<Long> mPinnedBlocks = new ArrayList<Long>();
+  private final List<Long> mReadingBlocks = new ArrayList<Long>();
```

Comment:
I prefer to rename `mReadingBlocks` to `mLockedBlocks` to be a little bit more general :)